### DISCUSSION TITLE
Lifecycle diagram does not correctly illustrate reducer/effect relationship

### DISCUSSION THREAD

**Discussion by DeborahK** [ngrx/platform#3095] [General]
## Other information:
The NgRx State Management Lifecycle does not correctly illustrate the relationship between reducers and effects as shown here: https://ngrx.io/generated/images/guide/store/state-management-lifecycle.png

Every action goes through every reducer and then goes through every effect. The diagram makes it appear as if an action is processed by one or the other.

## I would be willing to submit a PR for the docs :heart:

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[x ] No
(I'm not sure I could update an image?)

**Comment by timdeschryver** [comment]
Do you have something in mind how we could make this more clear @DeborahK ?

For the history of the current diagram (and some other versions) see https://github.com/ngrx/platform/issues/1981 .

**Comment by DeborahK** [comment]
It looks like the line from the reducer to the effect was there at one point and then later removed:
![image](https://user-images.githubusercontent.com/7987365/81001929-4ad5f680-8dfd-11ea-96e7-43cc7fb8f119.png)

Seems like for clarity that line should be there, maybe dashed as others suggested.

Otherwise it appears as if an action is going to the reducer OR the effect.

**Comment by DeborahK** [comment]
This is the diagram we've been using:
![image](https://user-images.githubusercontent.com/7987365/81002129-97b9cd00-8dfd-11ea-9aed-1037c3f12dbf.png)

**Comment by alex-okrushko** [comment]
Thanks @DeborahK 
I like your diagram. At the same time, I think it could be also misunderstood that reducers are invoking effects.

I typically use this one - it makes "stream of actions" more of a thing of its own:
<img width="914" alt="Screen Shot 2020-05-05 at 3 36 42 PM" src="https://user-images.githubusercontent.com/2830407/81108218-59e4a380-8ee6-11ea-97cb-36a45ba501be.png">

But at the same this I don't have any issues with the current diagram - it is meant to be very high level.

@jordanpowell88, I think you worked on the diagram. Any thoughts on it?

**Comment by jordanpowell88** [comment]
@alex-okrushko (sorry for the late response). My purpose in creating the diagram was to give a VERY high level picture for people new to NgRx. To give someone a general picture of what the different parts of NgRx do. As we all know, the more we work with NgRx we know that their are several different ways to building our apps and implementing features. 

One of the things I struggled with (as someone new to NgRx at the time) was It was hard to get strait answers about the "best and right" way to design the chart. Everyone would show me "their" chart they use which was slightly different than the previous persons chart.

Ultimately, I think the more we can get the community to commit to a "design" (regardless if it is "mine" or not) the better it will be the NgRx community and ultimately the easier it will be to get new people to migrate to NgRx.
